providers/amazon/src/airflow/providers/amazon/aws/hooks/ecs.py

```
@@ -29,6 +33,9 @@
 
 def should_retry(exception: Exception):
     """Check if exception is related to ECS resource quota (CPU, MEM)."""
+    if isinstance(exception, EcsCannotPullContainerError):
+        return False
+
     if isinstance(exception, EcsOperatorError):
```

IMO, let the `EcsCannotPullContainerError` error fail fast instead of retrying should be fine right ?

Based on the [Documentation - CannotPullContainer task errors in Amazon ECS](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_cannot_pull_image.html), it's more like configuration error from user instead of system instability.

cc @o-nikolas , @eladkal

---

I think there're cases suitable for retry with a reasonable wait time. e.g.,

[ERROR: toomanyrequests: Too Many Requests or You have reached your pull rate limit.](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_cannot_pull_image.html#container-pull-too-many-requests)